### PR TITLE
[Kunlun]add collective ops for multi XPU cards training and add Kunlun multi XPU cards CI

### PR DESCRIPTION
**Pull request by vslyu**
### PR types

New features
### PR changes

OPs
### Describe

- add collective ops for multi XPU cards training
add allreduce_sum/prod/max/min and reduce_sum/prod/max/min ops for XPU.

example:
fleetrun --xpus 0,1 example.py

```
import numpy as np
import paddle
from paddle.distributed import init_parallel_env

paddle.set_device('xpu:%d'%paddle.distributed.ParallelEnv().dev_id)
init_parallel_env()
if paddle.distributed.ParallelEnv().local_rank == 0:
    np_data = np.array([[4, 5, 6], [4, 5, 6]]).astype(np.float32)
else:
    np_data = np.array([[1, 2, 3], [1, 2, 3]]).astype(np.float32)
data = paddle.to_tensor(np_data)
paddle.distributed.reduce(data, 0)
out = data.numpy()
# [[5, 7, 9], [5, 7, 9]]
```
- add Kunlun multi XPU cards CI 
add dist xpu tests in python/paddle/fluid/tests/unittests/CMakeLists.txt.
UT name must contain "_xpu" to be executed, redefined example as following:
```
# dist xpu tests: 
if (WITH_XPU_BKCL)
    py_test(test_collective_reduce_api_xpu SRCS "test_collective_reduce_api.py")
    py_test(test_collective_allreduce_api_xpu SRCS "test_collective_allreduce_api.py")
endif()
```

### CODE REVIEW

File: paddle/fluid/operators/collective/c_allreduce_op.h [hunk 1]
```diff
@@ -20,11 +20,19 @@ limitations under the License. */
 #include "paddle/fluid/framework/lod_tensor.h"
 #include "paddle/fluid/framework/op_registry.h"
 
-#if defined(PADDLE_WITH_NCCL) || defined(PADDLE_WITH_RCCL)
+#if defined(PADDLE_WITH_NCCL) || defined(PADDLE_WITH_RCCL) || \
+    defined(PADDLE_WITH_XPU_BKCL)
 #include "paddle/fluid/platform/collective_helper.h"
+#endif
+
+#if (defined PADDLE_WITH_NCCL)
```
Comment (wangxicoding) [inline paddle/fluid/operators/collective/c_allreduce_op.h]: #if defined(PADDLE_WITH_NCCL) || defined(PADDLE_WITH_RCCL)

File: paddle/fluid/operators/collective/c_reduce_op.h [hunk 2]
```diff
@@ -24,10 +24,16 @@ limitations under the License. */
...
 #include "paddle/fluid/framework/op_registry.h"
 
-#if defined(PADDLE_WITH_NCCL) || defined(PADDLE_WITH_RCCL)
+#if defined(PADDLE_WITH_NCCL) || defined(PADDLE_WITH_RCCL) || \
+    defined(PADDLE_WITH_XPU_BKCL)
 #include "paddle/fluid/platform/collective_helper.h"
 #include "paddle/fluid/platform/nccl_helper.h"
 #endif
+
+#if defined(PADDLE_WITH_XPU_BKCL)
```
Comment (wangxicoding) [inline paddle/fluid/operators/collective/c_reduce_op.h:36]: #if defined(PADDLE_WITH_NCCL) || defined(PADDLE_WITH_RCCL)

File: paddle/fluid/operators/collective/c_allreduce_op.h (hunk 1 above)
Comment (vslyu) [reply paddle/fluid/operators/collective/c_allreduce_op.h]: done～

File: paddle/fluid/operators/collective/c_reduce_op.h (hunk 2 above)
Comment (vslyu) [reply paddle/fluid/operators/collective/c_reduce_op.h:36]: done～

**Review by QingshuChen — APPROVED** [review]
LGTM

**Review by wangxicoding — APPROVED** [review]

**Pull request merged**